src/components/property-tree/property-tree.tsx

```diff
@@ -0,0 +1,137 @@
+
+  const [treeViewData, setTreeViewData] = useState<TreeAsset[]>([]);
+
+  const addChildrenAssets = (): TreeAsset[] => {
+    const childrenAssets: TreeAsset[] = [];
+
+    if (childAssets) {
+      for (const [i, v] of childAssets.entries()) {
+        if (i < 5) {
+          childrenAssets.push(generateNode(v.assetAddress.addressLine1, [], v.id));
+        } else if (i === 5) {
+          childrenAssets.push({ title: "<< MORE.... >>", children: null });
+        } else {
+          break;
+        }
+      }
+    }
+
+    return childrenAssets;
+  };
+
+  const generateNode = (name: string, childList: TreeAsset[], id: string): TreeAsset => {
+    const node = (
+      <a className="lbh-link govuk-link" href={`/property/${id}`}>
+        {name}
+      </a>
+    );
+
+    return { title: node, children: childList, expanded: false };
+  };
+
+  const generatePrinciple = (asset: Asset, childNodes: TreeAsset[]): TreeAsset => {
+    if (asset.assetLocation?.parentAssets?.length) {
+      return {
+        title: <span>Principle</span>,
+        children: [
+          {
+            title: `${asset.assetAddress.addressLine1} (this asset)`,
+            children: childNodes,
+            expanded: true,
+          },
+        ],
+        expanded: true,
+      };
+    }
+
+    return {
+      title: <span>Hackney</span>,
+      children: [
+        {
+          title: `${asset.assetAddress.addressLine1} (this asset)`,
+          children: childNodes,
+          expanded: true,
+        },
+      ],
+      expanded: true,
+    };
+  };
+
+  const addParentsAndPrinciple = (
+    asset: Asset,
+    childNodes: TreeAsset[],
+    excludedTreeAssets: string,
+    principle: TreeAsset,
+  ) => {
+    const treeViewElements: TreeAsset[] = [];
+
+    if (asset.assetLocation?.parentAssets?.length) {
+      const validParents = asset.assetLocation.parentAssets.filter(
+        (el) => !excludedTreeAssets.includes(el.id),
+      );
+
+      for (const [i, v] of validParents.entries()) {
+        // Attach principle to last parent
+        if (i === validParents.length - 1) {
+          principle.title = (
+            <a className="lbh-link govuk-link" href={`/property/${v.id}`}>
+              {v.name}
+            </a>
+          );
+          treeViewElements.push(principle);
+        } else {
+          treeViewElements.push(generateNode(v.name, [], v.id));
+        }
+      }
+    } else {
+      treeViewElements.push(generatePrinciple(asset, childNodes));
+    }
+
+    setTreeViewData(treeViewElements);
+  };
+
+  const childNodes = addChildrenAssets();
+  const principle = generatePrinciple(asset, childNodes);
+
+  useEffect(() => {
+    addParentsAndPrinciple(asset, childNodes, excludedTreeAssets, principle);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const onChangeHandler = (treeData: TreeAsset[]) => {
+    setTreeViewData(treeData);
+  };
+
+  return (
+    <div>
```

Comment:
There is a lot of stuff going on in this component. Im finding it quite hard to understand it. I think if this was returned to in a couple of months, it would be dificult to make any changes.

I would suggest a few changes:
- move most of this logic into a custom hook. Only export what is needed. That simplifies the logic simply because the hook is essentially an interface. 
- make sure any variable/method name describes what the variable is and what its for
- add any comments where variable/methood names dont fully supply context

For example, on line 31, why is there a condition checking that `i` is less than 5? What is that checking for? And below that, the variable `v.id` is used. It is not clear to me what v contains.

Happy to discuss further

Reply:
Why not just extract the logic into a separate class/file? I'm not sure what the obsession with everything being a custom hook is. According to the [React documentation](https://react.dev/learn/reusing-logic-with-custom-hooks), they are for reusing logic that is used again and again. This is not being reused at all

Reply:
I agree on the re-usability, not being something we need in this case. What about creating a `property-tree-utils` file with the definitions of the methods used in `property-tree`? 

This would clean up the file a little bit, we can just call the methods from `property-tree`

Reply:
Angelo, take whatever approach you prefer. The important thing is the code is easy to read

Reply:
For reference, this was included in the hooks documentation page
![image](https://github.com/LBHackney-IT/mtfh-frontend-property/assets/88662046/1693544c-aa87-4136-9c60-d7231264faeb)